rskj-core/src/test/java/co/rsk/net/AsyncNodeBlockProcessorUnclesTest.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2017 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.net;
+
+import co.rsk.config.TestSystemProperties;
+import co.rsk.core.bc.BlockChainImpl;
+import co.rsk.net.simples.SimplePeer;
+import co.rsk.net.sync.SyncConfiguration;
+import co.rsk.net.utils.AsyncNodeBlockProcessorListener;
+import co.rsk.test.builders.BlockBuilder;
+import co.rsk.test.builders.BlockChainBuilder;
+import co.rsk.validators.BlockValidator;
+import co.rsk.validators.DummyBlockValidator;
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockHeader;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AsyncNodeBlockProcessorUnclesTest {
+
+    private static final long WAIT_TIME = Long.MAX_VALUE;
+
+    private BlockChainBuilder blockChainBuilder;
+    private BlockChainImpl blockChain;
+    private AsyncNodeBlockProcessor processor;
+    private AsyncNodeBlockProcessorListener listener;
+
+    @Before
+    public void setUp() {
+        blockChainBuilder = new BlockChainBuilder();
+        blockChain = blockChainBuilder.build();
+        listener = new AsyncNodeBlockProcessorListener();
+        processor = createAsyncNodeBlockProcessor(blockChain, listener);
+        processor.start();
+    }
+
+    @After
+    public void tearDown() throws InterruptedException {
+        processor.stopAndWait(WAIT_TIME);
+    }
```

Comment:
I liked these before/after, cant we do sthg similar with the other test class?

Reply:
Do you mean `AsyncNodeBlockProcessorTest` class? In this class `AsyncNodeBlockProcessor` is constructed differently for every test. Or did you mean something else?